include/libdnf5/base/transaction.hpp

```diff
@@ -163,6 +163,10 @@ class Transaction {
     /// Retrieve a list of the problems that occurred during `check_gpg_signatures` procedure.
     std::vector<std::string> get_gpg_signature_problems() const noexcept;
 
+    /// Serialize the transaction into a json data format which can be later loaded
+    /// into a `libdnf5::Goal` and replayed.
+    std::string serialize_to_json();
```

Comment:
The method is ok, but I have realized that there might be a white spot in our workflow if we want to download, test and store rpm path in the file.
Before download we can call `Command::goal_resolved()`, but there is no hook later.

Do you think whether it would be possible to return a different set of information if `serialize_to_json()`  as called before downloading of packages (without path to rpm) or after download (include paths of rpm)?

Reply:
> Do you think whether it would be possible to return a different set of information if `serialize_to_json()` as called before downloading of packages (without path to rpm) or after download (include paths of rpm)?

Yes, the `package_path` should be included only if the package is available locally.